packages/grafana-data/src/types/query.ts

```diff
@@ -52,3 +52,62 @@ export interface DataQuery {
    */
   datasource?: DataSourceRef | null;
 }
+
+/**
+ * Abstract representation of any label-based query
+ * @internal
+ */
+export interface AbstractQuery extends DataQuery {
+  labelMatchers: AbstractLabelMatcher[];
+}
+
+/**
+ * @internal
+ */
+export enum AbstractLabelOperator {
+  Equal = 'Equal',
+  NotEqual = 'NotEqual',
+  EqualRegEx = 'EqualRegEx',
+  NotEqualRegEx = 'NotEqualRegEx',
+}
+
+/**
+ * @internal
+ */
+export type AbstractLabelMatcher = {
+  name: string;
+  value: string;
+  operator: AbstractLabelOperator;
+};
+
+/**
+ * @internal
+ */
+export interface DataSourceWithQueryImportSupport<TQuery extends DataQuery> {
+  importAbstractQueries(labelBasedQuery: AbstractQuery[]): Promise<TQuery[]>;
+}
+
+/**
+ * @internal
+ */
+export interface DataSourceWithQueryExportSupport<TQuery extends DataQuery> {
+  exportToAbstractQueries(query: TQuery[]): Promise<AbstractQuery[]>;
+}
+
+/**
+ * @internal
+ */
+export const hasQueryImportSupport = <TQuery extends DataQuery>(
+  datasource: any
+): datasource is DataSourceWithQueryImportSupport<TQuery> => {
+  return (datasource as DataSourceWithQueryImportSupport<TQuery>).importAbstractQueries !== undefined;
+};
```

Comment:
Just a thought here - if we have `exportToAbstractQueries`, shouldn't this be `importFromAbstractQueries`?

Reply:
Makes sense, updated in cf4ea8e7fd168967b986d9223d67c4eb813ef5df